Sources/Target/Linux/Process.cpp

```diff
@@ -102,6 +105,52 @@ static pid_t blocking_waitpid(pid_t pid, int *status, int flags) {
   return ret;
 }
 
+ErrorCode Process::readMemory(Address const &address, void *data, size_t length,
+                              size_t *count) {
+#if defined(HAVE_PROCESS_VM_READV)
+  struct iovec local_iov = {data, length};
+  struct iovec remote_iov = {reinterpret_cast<void *>(address.value()), length};
+  auto id = _currentThread == nullptr ? _pid : _currentThread->tid();
+
+  ssize_t ret = process_vm_readv(id, &local_iov, 1, &remote_iov, 1, 0);
+  if (ret >= 0) {
+    if (count != nullptr) {
+      *count = ret;
+    }
+    return kSuccess;
+  } else {
+    DS2LOG(Debug, "process_vm_readv failed with errno=%s",
+           Stringify::Errno(errno));
+    // Fallback to super::readMemory, which uses ptrace(2).
+  }
+#endif
+
+  return super::readMemory(address, data, length, count);
+}
+
+ErrorCode Process::writeMemory(Address const &address, void const *data,
+                               size_t length, size_t *count) {
+#if defined(HAVE_PROCESS_VM_WRITEV)
+  struct iovec local_iov = {const_cast<void *>(data), length};
+  struct iovec remote_iov = {reinterpret_cast<void *>(address.value()), length};
+  auto id = _currentThread == nullptr ? _pid : _currentThread->tid();
```

Comment:
Don't we also discourage `auto` in cases where the type isn't explicitly present on the RHS (ie only allowed for constructors and casts)?